### PR TITLE
Feature/use bioio

### PR DESCRIPTION
**Pull request by saeliddp**
## What does this PR do?

Ports `aicsimageio` to `bioio`, which is recommended since only `bioio` is guaranteed to have dev support and updates moving forward. Also solves the issue we were having with `bfio` causing a segfault during testing. (see checks for verification). Also verified that with just the requirements in pyproject.toml, I can successfully run `pytest -k segmentation`.

Fixes [372](https://github.com/AllenCellModeling/cyto-dl/issues/372)

## Notes
- with `bioio`, you have to install plugins to deal with different image types--right now I have support for `ome-tiff` (non-tiled), `czi`, and `tiff` (non-globbed)
- I imagine we need to update the `requirements` directory after this update since we no longer need `aicsimage` and now need `bioio`
## Before submitting

- [x] Did you make sure **title is self-explanatory** and **the description concisely explains the PR**?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **test your PR locally** with `pytest` command?
- [x] Did you **run pre-commit hooks** with `pre-commit run -a` command?

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: cyto_dl/datamodules/czi.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 import numpy as np
 import pandas as pd
-from aicsimageio.aics_image import AICSImage
+from bioio import BioImage
```
Comment (saeliddp) [inline cyto_dl/datamodules/czi.py:6]: The API for `BioImage` and `AICSImage` is, as far as I can tell, identical. So, all we really need to do is swap out `AICSImage` for `BioImage` wherever we used it.

**Comment by saeliddp** [comment]
> I see the tests still failing in CI though - is that because `requirements.txt` still needs to be re-generated?

Ubuntu-latest 3.8 is failing because `bioio` requires python >= 3.9. The windows tests have been failing since before I started working on the dependency issues; I'm not sure what the cause of that is.

**Comment by benjijamorris** [comment]
Windows test are timing out at 70 minutes, it sounds like bioio will make tests faster + should fix this!

File: pyproject.toml [hunk 2]
```diff
@@ -44,7 +42,11 @@ dependencies = [
...
     "astropy>=5.2",
     "rich",
-    "boto3"
+    "boto3",
+    "bioio",
+    "bioio-czi",
+    "bioio-ome-tiff",
+    "bioio-tifffile"
 ]
 requires-python = ">=3.8,<3.11"
```
Comment (benjijamorris) [inline pyproject.toml:51]: Can you change this to >=3.9?

**Review by benjijamorris — CHANGES_REQUESTED** [review]
looks fantastic! I think we're ready to support >=3.9 only

File: pyproject.toml (hunk 2 above)
Comment (saeliddp) [reply pyproject.toml]: Done! [9d3f60](https://github.com/AllenCellModeling/cyto-dl/pull/376/commits/1b651d95ad40b128987a63e0f620c7981c9d3f60)

**Review by benjijamorris — APPROVED** [review]

**Pull request merged**